src/Parallel/MPIBasic.h

```diff
@@ -81,6 +87,14 @@ class MPIBasic
 		return m_rank;
 	}
 
+	/**
+	 * @return The rank of this process within this node
```

Comment:
Consider updating:
return the rank within the shared memory sub-communicator

Reply:
done